handout/architect/10_ways_to_misuse_angular.md

```diff
@@ -0,0 +1,103 @@
+    - No need to wrap templates in a `<div>`
+    - Use `:host` selector
+    - Use `@HostBinding` and `@HostListener`
+    - Subtle differences host metadata vs. `@Host*` decorators
+1. Avoiding observables
+    - Leverage observable operators
+    - Merge and transform data in streams rather than direct data mutation
+    - Don’t prematurely subscribe
+    - Leverage `async` pipe where appropriate
+1. Using the raw `Http` service
+    - Avoid using `Http` in components
+        - Use services instead
+        - Map out `res.json()` as part of a method call
+    - Work with data shapes that make sense for the application
+        - Don’t necessarily consume data as returned from a API
+        - Use a transformation layer to massage data into a shape that makes sense
+1. Not considering the injector tree when lazy loading modules
+    - DI may behave differently than expected
+    - Lazy loaded module providers will not be available until the module has been loaded in
+    - Lazy loaded module providers will not be globally available but rather scoped to the lazy module
+    - Prefer to provide at the root module level
+    - Leverage `forRoot` / `forChild` configuration
+1. Confusing the source of truth between router and application state
+    - State transitions should be links or router navigation calls
+    - Hydrate / mutate state based on router events
+        - Dispatch actions to store accordingly
+1. Excessive encapsulation of child components
+    - Use content projection where possible
+        - Keep a flatter component hierarchy
+            - Especially with forms
+    - Make more generic wrapper components
+    - Avoid excessive `@Input`/`@Output` chaining
+1. Using model-driven forms straight away
+    - Template-driven forms require less boilerplate, setup, and mental overhead
+        - More composable and reusable assuming a flat form hierarchy
+            - Reference previous point about content projection and flat form hierarchy
+        - Primary concern is the data model itself
+    - Model-driven forms tie a component to a specific schema or config
+        - More useful when working with complex dynamically generated forms
+            - Form models necessarily tend to be more generic
+            - Primary concern is the schema or display model of the form rather than the data model
+        - Making composable model-driven form components requires more work
+            - Passing `FormControl` references or implementing the `ControlValueAccessor` interface
+1. Using redux directly inside of components
+    - Select state using selector services
+    - Use action creator services to dispatch actions
+1. Direct DOM manipulation
+    - Prefer components or directives to achieve dynamic look and feel
+        - Prefer CSS for styling
+    - For dynamic templates, prefer structural directives
+    - Use `Renderer` if really necessary
+    - Only use `ElementRef` and DOM as a last resort
```

Comment:
Can you please include one example where someone would plausibly and correctly be driven to this last resort?